src/Workspaces/Core/Portable/Remote/RemoteHostSessionHelpers.cs

```diff
@@ -31,18 +31,15 @@ internal sealed class SessionWithSolution : IDisposable
 
                 // set connection state for this session.
                 // we might remove this in future. see https://github.com/dotnet/roslyn/issues/24836
-                await connection.InvokeAsync(
+                await connection.Target.InvokeAsync(
                     WellKnownServiceHubServices.ServiceHubServiceBase_Initialize,
                     new object[] { scope.SolutionInfo },
                     cancellationToken).ConfigureAwait(false);
 
-                return new SessionWithSolution(connection, scope);
+                return new SessionWithSolution(ref connection.Resource, scope);
```

Comment:
:memo: The use of a wrapper allows this function to avoid "taking ownership" of `connection` in error cases. The ownership doesn't transfer prior to this line, which means this method is no longer responsible for releasing the resource on error paths.

Reply:
shouldn't there be Move somewhere to move ownership?

Reply:
> shouldn't there be Move somewhere to move ownership?

Yes, see https://github.com/dotnet/roslyn/pull/26321#discussion_r183228151